packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -1133,6 +1154,36 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
 
         return { snapshotTree, blobs, reusedBlobs };
     }
+
+    private buildHierarchyToSummaryTree(flatTree: ITree): api.ISummaryTree {
```

Comment:
I wonder if we could switch away from using ITree.
I.e. today, snapshot indeed uses ITree structure. But in future this will change, so adding more dependencies on ITree will complicate my work :(
getSnapshotTree() right now is essentially the single place that knows about it and converts it to ISnaphotTree - format that runtime understands. That's the only place I need to change today (if you are interested - take a look at https://github.com/microsoft/FluidFramework/pull/4677).
So I wonder if we could call this API not in response of downloading snapshot, but in response of runtime code calling into getSnapshotTree, and use ISnapshotTree (result of that function) as an input to populate treesPathToSummaryTreeCache.
This would make code not dependent (more than it depends already today) on particular wire protocol, and possible be more reusable in the future (when r11s would have to go through same hurdles and would likely need to borrow most of this logic).

Reply:
Removed ITree. Using snapshot tree now.